ui/src/components/navigation/GlobalSearchModal.tsx

```diff
@@ -0,0 +1,347 @@
+              }}
+            >
+              {loading ? (
+                <Box sx={{ fontSize: '1rem', color: 'black', p: 2 }}>Loading...</Box>
+              ) : radioFilter === 'Songs' ? (
+                filterKeyword === '' ? (
+                  <SongSearchResult
+                    _id=""
+                    title=""
+                    keyword=""
+                    ref={(el) => (resultRefs.current[0] = el as HTMLDivElement)}
+                    isFocused={focusedIndex === 0}
+                    onClose={onClose}
+                  />
+                ) : filteredSongs && filteredSongs.length > 0 ? (
+                  filteredSongs.map((song, index) => (
+                    <SongSearchResult
+                      key={song._id}
+                      _id={song._id}
+                      title={song.title}
+                      keyword={filterKeyword}
+                      ref={(el) => (resultRefs.current[index] = el as HTMLDivElement)}
+                      isFocused={index === focusedIndex}
+                      onClose={onClose}
+                    />
+                  ))
+                ) : (
+                  <Box
+                    sx={{
+                      display: 'flex',
+                      flexDirection: 'row',
+                      justifyContent: 'flex-start',
+                      alignItems: 'center',
+                      width: '100%',
+                      p: '0.75rem',
+                    }}
+                  >
+                    <Typography variant="body1" color="#CAC4D0">
+                      No songs found
+                    </Typography>
+                  </Box>
+                )
+              ) : filterKeyword === '' ? (
+                <SetlistSearchResult
+                  _id=""
+                  name=""
+                  keyword={filterKeyword}
+                  ref={(el) => (resultRefs.current[0] = el as HTMLDivElement)}
+                  isFocused={focusedIndex === 0}
+                />
+              ) : filteredSetlists && filteredSetlists.length > 0 ? (
+                filteredSetlists.map((setlist, index) => (
+                  <SetlistSearchResult
+                    _id={setlist._id}
+                    name={setlist.name}
+                    keyword={filterKeyword}
+                    ref={(el) => (resultRefs.current[0] = el as HTMLDivElement)}
+                    isFocused={index === focusedIndex}
+                  />
+                ))
+              ) : (
+                <Box
+                  sx={{
+                    display: 'flex',
+                    flexDirection: 'row',
+                    justifyContent: 'flex-start',
+                    alignItems: 'center',
+                    width: '100%',
+                    p: '0.75rem',
+                  }}
+                >
+                  <Typography variant="body1" color="#CAC4D0">
+                    No setlists found
+                  </Typography>
+                </Box>
```

Comment:
minor issue here, having too much ternary operations is prone to bugs and error too. I think you can simplify the logic or just use multiple if statements as its also more readable. Like u can combine the filterKeyword and radioFilter by separating the song and setlist logic.
`radioFilter === 'Songs' && filterKeyword === '' `